SourceDistributions/Developer-Sandboxes/tsWxGTUI_PyVx/Python-3x-3.6.0/tsWxGTUI_Py3x-3.6.0/ManPages/tsManPagesLibGUI/tsWxCursor.man

Help on module tsWxCursor:

NAME
    tsWxCursor

FILE
    ./tsLibGUI/tsWxPkg/src/tsWxCursor.py

DESCRIPTION
    tsWxCursor.py - Class to establish a cursor from a small bitmap.
    The cursor is usually used for denoting where the mouse pointer
    is, with a picture that might indicate the interpretation of a
    mouse click.

CLASSES
    tsWxObject.Object(__builtin__.object)
        Cursor
    
    class Cursor(tsWxObject.Object)
     |  A cursor is a small bitmap usually used for denoting where the mouse
     |  pointer is, with a picture that might indicate the interpretation of
     |  a mouse click.
     |  
     |  A single cursor object may be used in many windows (any subwindow type).
     |  The wxWindows convention is to set the cursor for a window, as in X,
     |  rather than to set it globally as in MS Windows, although a global
     |  wx.SetCursor function is also available for use on MS Windows.
     |  
     |  Stock Cursor Names
     |  
     |  CURSOR_ARROW          A standard arrow cursor.
     |  CURSOR_ARROWWAIT      An arrow and an hourglass, (windows.)
     |  CURSOR_BLANK          Transparent cursor.
     |  CURSOR_BULLSEYE       Bullseye cursor.
     |  CURSOR_CHAR           Rectangular character cursor.
     |  CURSOR_CROSS          A cross cursor.
     |  CURSOR_DEFAULT        A platform dependent default cursor.
     |  CURSOR_HAND           A hand cursor.
     |  CURSOR_IBEAM          An I-beam cursor (vertical line).
     |  CURSOR_LEFT_BUTTON    Represents a mouse with the left button depressed.
     |  CURSOR_MAGNIFIER      A magnifier icon.
     |  CURSOR_MIDDLE_BUTTON  Represents a mouse with the middle button depressed.
     |  CURSOR_NO_ENTRY       A no-entry sign cursor.
     |  CURSOR_PAINT_BRUSH    A paintbrush cursor.
     |  CURSOR_PENCIL         A pencil cursor.
     |  CURSOR_POINT_LEFT     A cursor that points left.
     |  CURSOR_POINT_RIGHT    A cursor that points right.
     |  CURSOR_QUESTION_ARROW An arrow and question mark.
     |  CURSOR_RIGHT_ARROW    A standard arrow cursor pointing to the right.
     |  CURSOR_RIGHT_BUTTON   Represents a mouse with the right button depressed.
     |  CURSOR_SIZENESW       A sizing cursor pointing NE-SW.
     |  CURSOR_SIZENS         A sizing cursor pointing N-S.
     |  CURSOR_SIZENWSE       A sizing cursor pointing NW-SE.
     |  CURSOR_SIZEWE         A sizing cursor pointing W-E.
     |  CURSOR_SIZING         A general sizing cursor.
     |  CURSOR_SPRAYCAN       A spraycan cursor.
     |  CURSOR_WAIT           A wait cursor.
     |  CURSOR_WATCH          A watch cursor.
     |  
     |  Method resolution order:
     |      Cursor
     |      tsWxObject.Object
     |      __builtin__.object
     |  
     |  Methods defined here:
     |  
     |  IsOk(self)
     |  
     |  Ok(self)
     |  
     |  __init__(self, cursorName, type, hotSpotX=0, hotSpotY=0)
     |      Associate a curses cursor visibility mode with a wxPython cursor
     |      name instead of the following.
     |      
     |      Construct a Cursor from a file. Specify the type of file using
     |      wx.BITMAP_TYPE* constants, and specify the hotspot if not using
     |      a .cur file. :see: Alternate constructors wx.StockCursor,
     |      `wx.CursorFromImage`
     |  
     |  __nonzero__(self)
     |  
     |  ----------------------------------------------------------------------
     |  Methods inherited from tsWxObject.Object:
     |  
     |  Destroy(self)
     |      Deletes the C++ object this Python object is a proxy for.
     |  
     |  GetClassInfo(self)
     |      Return information about the class.
     |  
     |  GetClassName(self)
     |      Returns the class name of the C++ class using wxRTTI.
     |  
     |  IsSameAs(self, p)
     |      For wx.Objects that use C++ reference counting internally, this
     |      method can be used to determine if two objects are referencing
     |      the same data object.
     |  
     |  tsBeginClassRegistration(self, theClass, applicationId)
     |      Establish connection with application logger. Log the beginning of
     |      class registration.
     |      
     |      Establish application access to display top-level window and to
     |      curses screen and windows.
     |  
     |  tsEndClassRegistration(self, theClass)
     |      Log the ending of class registration.
     |  
     |  tsGetClassNameIndent(self, theClassName)
     |      Return the hierarchical level associated with the specified class
     |      for use in indenting the logging of class initialization entry and
     |      exit.
     |  
     |  tsGetTerminalPixelRectangle(self)
     |      Returns the bounding rectangle the client area of the display,
     |      i.e., without taskbars and such.
     |  
     |  tsGetTheLogger(self)
     |      Return the logger instance.
     |  
     |  tsGetTheTerminal(self, theClass)
     |      Return the terminal instance.
     |  
     |  tsGetTheTerminalScreen(self, theClass)
     |      Return the screen instance.
     |  
     |  tsGetThisOwn(self)
     |      Return MembershipFlag.
     |  
     |  tsInstallTheLoggerAccess(self, indent, theClass, applicationId)
     |      Establish connection with application logger.
     |  
     |  tsInstallTheTerminalAccess(self, indent, theClass, applicationId)
     |      Establish those display areas reserved for task bar and stdio
     |      redirection. Create links to the internal information needed
     |      to monitor and control the physical screen and virtual windows
     |      of the Graphical Text User Interface.
     |  
     |  tsRegisterClassNameAndMembershipFlag(self, theClass)
     |      Record Class Name and Membership Flag instance variables.
     |  
     |  tsUpdateEventAndAcceleratorTables(self, event, handler, source, id=-1, id2=-1, useSystemEventTable=False)
     |      Bind with event and accelerator System and User tables.
     |      
     |      event: One of the EVT_* objects that specifies the
     |             the type of event to bind.
     |      
     |      handler: A callable object to be invoked when the event is
     |               delivered to self.  Pass None to disconnect an
     |               event handler.
     |      
     |      source: Sometimes the event originates from a different window
     |              than self, but you still want to catch it in self.  (For
     |              example, a button event delivered to a frame.)  By passing
     |              the source of the event, the event handling system is able
     |              to differentiate between the same event type from different
     |              controls.
     |      
     |      id: Used to specify the event source by ID instead of instance.
     |      
     |      id2: Used when it is desirable to bind a handler to a range of ids,
     |           such as with EVT_MENU_RANGE.
     |  
     |  ----------------------------------------------------------------------
     |  Static methods inherited from tsWxObject.Object:
     |  
     |  tsGetCurrentId()
     |      Return latest ID.
     |  
     |  tsGetTheId(mySelf)
     |      Return the ID associated with this class instance.
     |  
     |  tsNewId()
     |      Create a unique ID.
     |      
     |      Remarks:
     |      Various controls and other parts of wxWidgets need an ID. Sometimes
     |      the ID may be directly provided by the user or have a predefined
     |      value, such as wxID_OPEN. Often, however, the value of the ID is
     |      unimportant and is created automatically by calling
     |      wxWindow::NewControlId or by passing wxID_ANY as the ID of an
     |      object.
     |      
     |      There are two ways to generate an ID. One way is to start at a
     |      negative number, and for each new ID, return the next smallest
     |      number. This is fine for systems that can use the full range of
     |      negative numbers for IDs, as this provides more than enough IDs
     |      and it would take a very very long time to run out and wrap
     |      around. However, some systems cannot use the full range of the
     |      ID value. Windows, for example, can only use 16 bit IDs, and
     |      only has about 32000 possible automatic IDs that can be
     |      generated by wxWindow::NewControlId. If the program runs long
     |      enough, depending on the program itself, using this first method
     |      would cause the IDs to wrap around into the positive ID range
     |      and cause possible clashes with any directly specified ID values.
     |      
     |      The other way is to keep track of the IDs returned by
     |      wxWindow::NewControlId and do not return them again until the
     |      ID is completely free and not being used by any other objects.
     |      This will make sure that the ID values do not clash with one
     |      another. This is accomplished by keeping a reference count for
     |      each of the IDs that can possibly be returned by
     |      wxWindow::NewControlId. Other IDs are not reference counted.
     |  
     |  tsRegisterId(usedId)
     |      Adjust latest ID to reflect specified new one.
     |  
     |  ----------------------------------------------------------------------
     |  Data descriptors inherited from tsWxObject.Object:
     |  
     |  ClassName
     |      Returns the class name of the C++ class using wxRTTI.
     |  
     |  __dict__
     |      dictionary for instance variables (if defined)
     |  
     |  __weakref__
     |      list of weak references to the object (if defined)
     |  
     |  thisown
     |      Return MembershipFlag.
     |  
     |  ----------------------------------------------------------------------
     |  Data and other attributes inherited from tsWxObject.Object:
     |  
     |  CurrentId = 100
     |  
     |  PyApp_EventQueue = None
     |  
     |  PyApp_Object = None
     |  
     |  TheDisplay = None
     |  
     |  TheLogger = None
     |  
     |  TheTerminal = None
     |  
     |  TheTerminalScreen = None

DATA
    DEBUG = True
    VERBOSE = True
    __authors__ = 'Richard S. Gordon'
    __copyright__ = 'Copyright (c) 2007-2013 Richard S. Gordon.\n\t\tAll r...
    __credits__ = '\n\n  Credits: \n\n\t  tsLibGUI Import & Application .....
    __date__ = '04/01/2013'
    __header__ = '\n\ntsWxCursor, v0.0.0 (build 04/01/2013)\n\n  Autho... ...
    __license__ = 'GNU General Public License, Version 3, 29 June 2007'
    __line1__ = 'tsWxCursor, v0.0.0 (build 04/01/2013)'
    __line2__ = 'Author(s): Richard S. Gordon'
    __line3__ = 'Copyright (c) 2007-2013 Richard S. Gordon.\n\t\tAll right...
    __line4__ = 'GNU General Public License, Version 3, 29 June 2...U Gene...
    __title__ = 'tsWxCursor'
    __version__ = '0.0.0'
    cursesInvisibleCursorMode = 0
    cursesUnderlineCursorMode = 1
    cursesVeryVisibleCursorMode = 2
    mainTitleVersionDate = 'tsWxCursor, v0.0.0 (build 04/01/2013)'

VERSION
    0.0.0

DATE
    04/01/2013

CREDITS
    
    
      Credits: 
    
    	  tsLibGUI Import & Application Launch Features: 
    	  Copyright (c) 2007-2009 Frederick A. Kier.
    			All rights reserved.
    
    	  Python Curses Module API & Run Time Library Features:
    	  Copyright (c) 2001-2013 Python Software Foundation.
    			All rights reserved.
    	  PSF License Agreement for Python 2.7.3 & 3.3.0
    
    	  wxWidgets (formerly wxWindows) & wxPython API Features:
    	  Copyright (c) 1992-2008 Julian Smart, Robert Roebling,
    			Vadim Zeitlin and other members of the 
    			wxWidgets team.
    			All rights reserved.
    	  wxWindows Library License
    
    	  nCurses API & Run Time Library Features:
    	  Copyright (c) 1998-2011 Free Software Foundation, Inc.
    			All rights reserved.
    	  GNU General Public License, Version 3, 29 June 2007




Documents/NEWS.txt
#-----------------------------------------------------------
#"Time-stamp: <07/08/2015  4:48:41 AM rsg>"
#-----------------------------------------------------------

========================= NEWS.txt =========================

   +----+----+  TeamSTARS "tsWxGTUI_PyVx" Toolkit
   | ts | Wx |      with Python 2x & Python 3x based
   +----+----+         Command Line Interface (CLI)
   | G T U I |      and "Curses"-based "wxPython"-style, 
   +---------+         Graphical-Text User Interface (GUI)
 
   Get that cross-platform, pixel-mode "wxPython" feeling
   on character-mode 8-/16-color (xterm-family) & non-color
   (vt100-family) terminals and terminal emulators.

   You can find this and other plain-text files in the
   Toolkit subdirectory named:

       "./<Toolkit Recipient's Repository>/Documents".

       <Your Working Repository>
       (e.g. "tsWxGTUI_PyVx_Repository") 
         |
         |
         +-- ["Documents"]

=========================== News ===========================

The pre-alpha release of the TeamSTARS "tsWxGTUI_PyVx-0.0.0"
Toolkit is now available. It includes the following two
versions abd their associated variants (Python Distribution
Utilities ("Distutils") installable "Site-Packages" and non-
installable "Developer-Sandboxes" for enhancement and bug
fix development and experimentation without interference
with the already installed Site-Packages):

1. The TeamSTARS "tsWxGTUI_Py2x-0.0.0" Toolkit is released
   for users of the popular, high-level, cross-platform,
   second generation Python 2.x programming language which
   is widely available.

   A wiki posting by the Python Software Foundation announc-
   ed that Python 3.0 was released in 2008. The final 2.x
   version 2.7 release came out in mid-2010, with a state-
   ment of extended support for this end-of-life release.
   The 2.x branch will see no new major releases after that.

   The source code and documentation is available for down-
   load at:

      http://pypi.python.org/pypi.

   It supports development and operation of:

   a. Command Line User Interface (CLI) application
      programs implemented in Python 2.0.0-2.7.9.

   b. Graphical-style User Interface (GUI) application
      programs implemented in Python 2.6.4-2.7.9.

2. The TeamSTARS "tsWxGTUI_Py3x-0.0.0" Toolkit is released
   for users of the popular, high-level, cross-platform,
   third generation Python 3.x programming language which
   is growing in availability.

   A wiki posting by the Python Software Foundation announc-
   ed that Python 3.0 was released in 2008. 3.x is under
   active development and has already seen over five years
   of stable releases, including version 3.3 in 2012 and
   3.4 in 2014. This means that all recent standard library
   improvements, for example, are only available by default
   in Python 3.x.

   Guido van Rossum (the original creator of the Python
   language) decided to clean up Python 2.x properly,
   with less regard for backwards compatibility than is
   the case for new releases in the 2.x range. The most
   drastic improvement is the better Unicode support
   (with all text strings being Unicode by default) as
   well as saner bytes/Unicode separation.

   Besides, several aspects of the core language (such as
   print and exec being statements, integers using floor
   division) have been adjusted to be easier for newcomers
   to learn and to be more consistent with the rest of the
   language, and old cruft has been removed (for example,
   all classes are now new-style, "range()" returns a memory
   efficient iterable, not a list as in 2.x).

   The source code and documentation is available for down-
   load at:

      http://pypi.python.org/pypi.

   It supports development and operation of:

   a. Command Line User Interface (CLI) application
      programs implemented in Python 3.0.0-3.4.2.

   b. Graphical-style User Interface (GUI) application
      programs implemented in Python 3.1.5-3.4.2.

======================= End-Of-File ========================


SourceDistributions/Developer-Sandboxes/tsWxGTUI_PyVx/Python-2x/tsWxGTUI_Py2x/tsDemoArchive/tsTutorialDeveloperSandbox2x/tsLibGUI/tsWxPkg/src/tsWxCursor.py
#! /usr/bin/env python
# "Time-stamp: <10/25/2013  6:29:05 AM rsg>"
'''
tsWxCursor.py - Class to establish a cursor from a small bitmap.
The cursor is usually used for denoting where the mouse pointer
is, with a picture that might indicate the interpretation of a
mouse click.
'''
#################################################################
#
# File: tsWxCursor.py
#
# Purpose:
#
#    Class to establish a cursor from a small bitmap. The cursor
#    is usually used for denoting where the mouse pointer is,
#    with a picture that might indicate the interpretation of a
#    mouse click.
#
# Usage (example):
#
#    # Import
#
#    from tsWxCursor import Cursor
#
# Requirements:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Capabilities:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Limitations:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Notes:
#
#    None
#
# Methods:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Classes:
#
#    wxPython 2.8.9.2 (New wxPyDocs)/wxWidgets 2.9.2 (Ref. Man.).
#
# Modifications:
#
#    None.
#
# ToDo:
#
#    None.
#
#################################################################

__title__     = 'tsWxCursor'
__version__   = '0.0.0'
__date__      = '04/01/2013'
__authors__   = 'Richard S. Gordon'
__copyright__ = 'Copyright (c) 2007-2013 ' + \
                '%s.\n\t\tAll rights reserved.' % __authors__
__license__   = 'GNU General Public License, ' + \
                'Version 3, 29 June 2007'
__credits__   = '\n\n  Credits: ' + \
                '\n\n\t  tsLibGUI Import & Application Launch Features: ' + \
                '\n\t  Copyright (c) 2007-2009 Frederick A. Kier.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\n\t  Python Curses Module API & ' + \
                'Run Time Library Features:' + \
                '\n\t  Copyright (c) 2001-2013 ' +\
                'Python Software Foundation.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  PSF License Agreement for Python 2.7.3 & 3.3.0' + \
                '\n\n\t  wxWidgets (formerly wxWindows) & ' + \
                'wxPython API Features:' + \
                '\n\t  Copyright (c) 1992-2008 Julian Smart, ' + \
                'Robert Roebling,' + \
                '\n\t\t\tVadim Zeitlin and other members of the ' + \
                '\n\t\t\twxWidgets team.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  wxWindows Library License' + \
                '\n\n\t  nCurses API & Run Time Library Features:' + \
                '\n\t  Copyright (c) 1998-2011 ' + \
                'Free Software Foundation, Inc.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  GNU General Public License, ' + \
                'Version 3, 29 June 2007'

__line1__ = '%s, v%s (build %s)' % (__title__, __version__, __date__)
__line2__ = 'Author(s): %s' % __authors__
__line3__ = '%s' % __copyright__

if len( __credits__) == 0:
    __line4__ = '%s' % __license__
else:
    __line4__ = '%s%s' % (__license__, __credits__)

__header__ = '\n\n%s\n\n  %s\n  %s\n  %s\n' % (__line1__,
                                               __line2__,
                                               __line3__,
                                               __line4__)

mainTitleVersionDate = __line1__

import time

import tsExceptions as tse
import tsLogger
import tsWxGlobals as wx
import tsWxGraphicalTextUserInterface as tsGTUI
from tsWxObject import Object

#---------------------------------------------------------------------------

##DEBUG = True
DEBUG = wx.Debug_GUI_Launch | \
        wx.Debug_GUI_Progress | \
        wx.Debug_GUI_Termination | \
        wx.Debug_GUI_Exceptions

##VERBOSE = True
VERBOSE = wx.Debug_GUI_Configuration

cursesInvisibleCursorMode = 0
cursesUnderlineCursorMode = 1
cursesVeryVisibleCursorMode = 2

#---------------------------------------------------------------------------

class Cursor(Object):
    '''
    A cursor is a small bitmap usually used for denoting where the mouse
    pointer is, with a picture that might indicate the interpretation of
    a mouse click.

    A single cursor object may be used in many windows (any subwindow type).
    The wxWindows convention is to set the cursor for a window, as in X,
    rather than to set it globally as in MS Windows, although a global
    wx.SetCursor function is also available for use on MS Windows.

    Stock Cursor Names

    CURSOR_ARROW          A standard arrow cursor.
    CURSOR_ARROWWAIT      An arrow and an hourglass, (windows.)
    CURSOR_BLANK          Transparent cursor.
    CURSOR_BULLSEYE       Bullseye cursor.
    CURSOR_CHAR           Rectangular character cursor.
    CURSOR_CROSS          A cross cursor.
    CURSOR_DEFAULT        A platform dependent default cursor.
    CURSOR_HAND           A hand cursor.
    CURSOR_IBEAM          An I-beam cursor (vertical line).
    CURSOR_LEFT_BUTTON    Represents a mouse with the left button depressed.
    CURSOR_MAGNIFIER      A magnifier icon.
    CURSOR_MIDDLE_BUTTON  Represents a mouse with the middle button depressed.
    CURSOR_NO_ENTRY       A no-entry sign cursor.
    CURSOR_PAINT_BRUSH    A paintbrush cursor.
    CURSOR_PENCIL         A pencil cursor.
    CURSOR_POINT_LEFT     A cursor that points left.
    CURSOR_POINT_RIGHT    A cursor that points right.
    CURSOR_QUESTION_ARROW An arrow and question mark.
    CURSOR_RIGHT_ARROW    A standard arrow cursor pointing to the right.
    CURSOR_RIGHT_BUTTON   Represents a mouse with the right button depressed.
    CURSOR_SIZENESW       A sizing cursor pointing NE-SW.
    CURSOR_SIZENS         A sizing cursor pointing N-S.
    CURSOR_SIZENWSE       A sizing cursor pointing NW-SE.
    CURSOR_SIZEWE         A sizing cursor pointing W-E.
    CURSOR_SIZING         A general sizing cursor.
    CURSOR_SPRAYCAN       A spraycan cursor.
    CURSOR_WAIT           A wait cursor.
    CURSOR_WATCH          A watch cursor.
    '''

    def __init__(self, cursorName, type, hotSpotX=0, hotSpotY=0):
        '''
        Associate a curses cursor visibility mode with a wxPython cursor
        name instead of the following.
 
        Construct a Cursor from a file. Specify the type of file using
        wx.BITMAP_TYPE* constants, and specify the hotspot if not using
        a .cur file. :see: Alternate constructors wx.StockCursor,
        `wx.CursorFromImage`
        '''
        theClass = 'Cursor'

        wx.RegisterFirstCallerClassName(self, theClass)

        Object.__init__(self)
        try:
            self.theFirstCallerName != theClass
        except AttributeError:
            self.theFirstCallerName = theClass

        self.tsBeginClassRegistration(theClass, wx.ID_ANY)
        try:

            self.logger = tsLogger.TsLogger(threshold=tsLogger.ERROR,
                                            start=time.time(),
                                            name=tsLogger.StandardOutputFile)

        except AttributeError, e:

            self.logger = None
            msg = 'tsWxCursor.__init__: Exception = %s' % e
            raise tse.ProgramException(tse.APPLICATION_TRAP, msg)

        if tsGTUI:

            self.ts_CursorName = cursorName
            self.ts_CursorHotSpotX = hotSpotX
            self.ts_CursorHotSpotY = hotSpotY

            if cursorName == wx.CURSOR_BLANK_NAME:

                self.ts_CursorType = cursesInvisibleCursorMode

            elif cursorName == wx.CURSOR_CHAR_NAME:

                self.ts_CursorType = cursesVeryVisibleCursorMode

            else:

                # cursorName == wx.CURSOR_DEFAULT_NAME
                self.ts_CursorType = cursesUnderlineCursorMode

            tsGTUI.tsSetCursesCursor(self.ts_CursorType)

        else:
            self.ts_CursorName = cursorName
            self.ts_CursorType = type
            self.ts_CursorHotSpotX = hotSpotX
            self.ts_CursorHotSpotY = hotSpotY

        self.tsEndClassRegistration(theClass)
 
##    def __del__(self):
##        '''
##        '''
##        pass

    def __nonzero__(self):
        '''
        '''
        return (True)

    def IsOk(self):
        '''
        '''
        return (True)

    def Ok(self):
        '''
        '''
        return (True)

#---------------------------------------------------------------------------

if __name__ == '__main__':

    print(__header__)
